NetKet/Output/json_output_writer.hpp

```diff
@@ -96,7 +96,7 @@ class JsonOutputWriter {
    * frequency.
    */
   template <class State>
-  void WriteState(int iteration, const State& state) {
+  void WriteState(int iteration, State state) {
```

Comment:
Why copy the state? Nothing in WriteState retains the reference so it is perfectly fine to just use a const reference.

Reply:
If this is changed back to take a const reference, you will have to replace `psi_` by `*psi_` when it is called with a machine from VMC:
https://github.com/netket/netket/blob/e90628faed38c6f22ee6691588ae8b84840f210d/NetKet/GroundState/variational_montecarlo.hpp#L401

Reply:
That's the point, psi_ here is a shared pointer, I can't revert it to a reference

Reply:
Sorry I misunderstood your comment. I can change that

Reply:
Should be good now